_applications/equoid.adoc

```diff
@@ -0,0 +1,219 @@
+
+== Software Components
+
+We chose to use Artemis for the message broker, with the expectation it would be scalable in terms of number of streams as well as volume of traffic. In the same vein, we utilized Spark for our microservice requiring scalability, namely the data handler. Since we wished to take advantage of extant Spark libraries, and Spark's `CMSketch` object is only (as of this writing) available in Java and Scala, we chose to implement the handler in Scala. In the interest of uniformity, the sample data publisher is also written in Scala. Furthermore, the AMQP Spark connector used by the data handler was developed by Paolo Patierno at Red Hat, with more information https://radanalytics.io/examples/amqpstreaming[this tutorial] and associated https://github.com/redhat-iot/amqp-spark-demo[upstream project].
+
+pass:[<img src="/assets/equoid/Equoid-Software-Components.png" alt="Alt text" class="img-responsive arch" width="800px">]
+ 
+[[installation]]
+== Installation
+
+Equoid consists of a number of microservices which require deployment to an OpenShift instance. In order to get you started as swiftly as possible, in this section a walkthrough of installing and configuring the Equoid services is given. 
+
+[[prerequisites]]
+=== Prerequisites
+
+* An available OpenShift instance.
+* A terminal with the OpenShift (`oc`) command and active login session are available.
+* An OpenShift project you own.
+
+[[procedure]]
+=== Procedure
+
+. Install Image Streams and Templates
++
+Depending on your OpenShift instance, some of the image streams necessary for Equoid may not be available. To be certain you have what's necessary to build Equoid, run the following sequence of commands:
++
+....
+oc create -f https://raw.githubusercontent.com/radanalyticsio/equoid-openshift/master/fabric8-image-streams.json
+oc create -f https://radanalytics.io/resources.yaml
+oc create -f https://raw.githubusercontent.com/infinispan/infinispan-openshift-templates/master/templates/infinispan-ephemeral.json
+....
++
+These provide the Fabric8 fuse-java image stream for the data-publisher service, the radanalytics.io Oshinko resources for facilitating Spark cluster creation and deployment as well as S2I for the data-handler, and the Infinispan template for the caching microservice, respectively. 
+
+. Launch Artemis
++
+We'll get Artemis up and running, as it is necessary for both the publisher and handler:
++
+....
+oc create -f https://raw.githubusercontent.com/radanalyticsio/equoid-openshift/master/artemis-rc.yaml
+....
+
+. Launch Infinispan Cache
++
+We'll want our caching layer in place before we begin streaming and ingesting. The following will get an Infinispan pod running. `APPLICATION_USER`, `APPLICATION_PASSWORD`, `MANAGEMENT_USER` and `MANAGEMENT_PASSWORD` should be changed to your desired values. 
++
```

Comment:
does it matter what the values for `APPLICATION_USER`, `APPLICATION_PASSWORD`, `MANAGEMENT_USER` and `MANAGEMENT_PASSWORD` are?

if not, i think we should have some sensible defaults in the example command.

Reply:
The credentials are required by the template, but aren't needed by any of Equoid's components. If one wants to look at the data in the cache they need to log in to the management console, that can be handy for testing. 

Given your overall comments in the change request I figured I'd populate all the values with sensible defaults. Was thinking "radley" for all of them.